node/src/Router.ts

```diff
@@ -140,16 +144,21 @@ export class Router extends EnhancedEventEmitter
 	// DataProducers map.
 	readonly #dataProducers: Map<string, DataProducer> = new Map();
 
-	// Router to PipeTransport map.
-	readonly #mapRouterPipeTransports: Map<Router, PipeTransport[]> = new Map();
-
-	// AwaitQueue instance to make pipeToRouter tasks happen sequentially.
-	readonly #pipeToRouterQueue =
-		new AwaitQueue({ ClosedErrorClass: InvalidStateError });
+	// Map of PipeTransport pair Promises indexed by a string computed with the
+	// ids of both involved Routers.
+	readonly #mapRouterPairPipeTransportPairPromise:
+		Map<string, Promise<PipeTransportPair>> = new Map();
```

Comment:
I think this can be simplified by replacing `Map<string, Promise<PipeTransportPair>>` with `Map<Router, Promise<PipeTransportPair>>` as I originally suggested, where `Router` is the remote router in relation to router on which this property is being accessed. Then `getPipeTransportPairKey()` will become unnecessary.

Reply:
On it.

Reply:
Done. Rather than `Map<Router, Promise<PipeTransportPair>>` I've used `Map<string, Promise<PipeTransportPair>>`, where the key is the `id` of the router in which `pipeToRouter()` was called.